azurerm/resource_arm_app_service.go

```diff
@@ -279,6 +281,16 @@ func resourceArmAppServiceCreate(d *schema.ResourceData, meta interface{}) error
 
 	d.SetId(*read.ID)
 
+	authSettings := azure.ExpandAppServiceAuthSettings(d.Get("auth_settings"))
+
+	auth := web.SiteAuthSettings{
+		ID:                         read.ID,
+		SiteAuthSettingsProperties: &authSettings}
+
+	if _, err := client.UpdateAuthSettings(ctx, resGroup, name, auth); err != nil {
+		return err
```

Comment:
can we wrap this error?

```suggestion
		return fmt.Errorf("Error updating auth settings for App Service %q (Resource Group %q): %+s", name, resGroup, err)
```